frontend/api.js

```diff
@@ -1,10 +1,33 @@
-import { ApolloClient, InMemoryCache, gql } from "@apollo/client";
+import {
+  ApolloClient,
+  InMemoryCache,
+  gql,
+  createHttpLink,
+} from "@apollo/client";
+import { setContext } from "@apollo/client/link/context";
 
 const API_URL = "https://api.lens.dev";
 
+const authLink = setContext((_, { headers }) => {
+  const token = window.localStorage.getItem("TOKEN");
+  const tokenRepresh = window.localStorage.getItem("TOKENREFRESH");
+  return {
+    headers: {
+      ...headers,
+      authorization: token ? `Bearer ${token}` : "",
+      "refresh-token": `${tokenRepresh}`,
+    },
+  };
+});
```

Comment:
Here is an example of setting up the Apollo client authlink from lenster...
https://github.com/lensterxyz/lenster/blob/08bf0da3efc8eb60d4cdb8eda55aa390be2499d9/apps/web/src/apollo.ts#L52-L102

So a couple of things...
- The logic for the authlink should be wrapped up in the `ApolloLink` class from the `'@apollo/client'` npm module
- Inside the function you pass to the `ApolloLink` class constructor, the logic takes care of checking to see when the current access token will expire. If the token has expired the refresh mutation is called to get a fresh new access token using the refresh token...
https://github.com/lensterxyz/lenster/blob/08bf0da3efc8eb60d4cdb8eda55aa390be2499d9/apps/web/src/apollo.ts#L60-L83

And all the above logic will get called on every apollo client request acting like a middleware. So no matter where you call out to the lens graphql api, this middleware will make sure you have a valid access token to reach out to the api.

Reply:
Also on a side note it looks like lenster uses `graphql-codegen` module from npm ( https://the-guild.dev/graphql/codegen )...
https://github.com/lensterxyz/lenster/blob/1155dce116de687efd884c29a1ecb36a6388e159/packages/lens/package.json#L14
It looks pretty awesome. Just from define graphql queries in .graphql files it can generate react-apollo hooks to use in any react functional component.